tests/rptest/tests/crl_test.py

```diff
@@ -0,0 +1,279 @@
+
+        RPC_TLS_CONFIG = dict(
+            enabled=True,
+            require_client_auth=True,
+            key_file=RedpandaService.TLS_SERVER_KEY_FILE,
+            cert_file=RedpandaService.TLS_SERVER_CRT_FILE,
+            truststore_file=RedpandaService.TLS_CA_CRT_FILE,
+            crl_file=RedpandaService.TLS_CA_CRL_FILE,
+        )
+
+        self.redpanda.stop()
+        self.redpanda.start(node_config_overrides={
+            n: dict(rpc_server_tls=RPC_TLS_CONFIG)
+            for n in self.redpanda.nodes
+        })
+
+        def cluster_health(node: ClusterNode,
+                           healthy: bool = True,
+                           noisy: bool = False,
+                           reasons: list[str] = []):
+            assert not (healthy and len(reasons)
+                        > 0), "Reasons apply only to unhealthy clusters"
+            ch = self.admin.get_cluster_health_overview(node)
+            if noisy:
+                self.logger.debug(f"health overview: {json.dumps(ch)}")
+            return ch['is_healthy'] is healthy and all(
+                r in ch['unhealthy_reasons'] for r in reasons)
+
+        wait_until(lambda: cluster_health(node),
+                   timeout_sec=10,
+                   backoff_sec=0.2,
+                   err_msg="Cluster did not become healthy")
+
+        self.logger.debug(
+            f"Now revoke the broker cert and push to {node.account.hostname}")
+
+        broker_cert = self.provider.broker_certs[0]
+
+        assert node.account.hostname in broker_cert.crt, f"Cert order mismatch: {broker_cert.crt}"
+
+        self.tls.revoke_cert(broker_cert)
+        self.redpanda.write_crl_file(node, self.tls.ca)
+
+        self.redpanda.restart_nodes([node])
+        other_node = self.redpanda.nodes[1]
+
+        self.logger.debug(
+            f"{node.account.hostname} should appear 'down' to the rest of the cluster"
+        )
+
+        wait_until(lambda: cluster_health(
+            other_node, healthy=False, noisy=True, reasons=['nodes_down']),
+                   timeout_sec=10,
+                   backoff_sec=0.2,
+                   err_msg="Cluster did not become unhealthy")
+
+    @cluster(num_nodes=3)
+    def test_noncogent(self):
+        node = self.redpanda.nodes[0]
+        self.rpk.list_schemas()
+
+        other_tls = tls.TLSCertManager(self.logger)
+        some_cert = other_tls.create_cert(socket.gethostname(),
+                                          common_name='foo',
+                                          name='foo')
+        other_tls.revoke_cert(some_cert)
+        for n in self.redpanda.nodes:
+            self.redpanda.write_crl_file(n, other_tls.ca)
+
+        self.logger.debug(
+            "TODO: As implemented, this quietly breaks everything. Behavior should be both graceful and configurable."
+        )
+        try:
+            self.rpk.list_schemas()
+        except:
+            pass
```

Comment:
Yeah, I will document it on the ticket I think. Nothing worth reporting in tests since we'll address it all next week.

Reply:
You know what, I think this is basically all we get

```
DEBUG 2024-05-30 21:02:59,542 [shard 0:main] rpc - transport.cc:400 - could not parse header from server: {host: docker-rp-3, port: 33145}
DEBUG 2024-05-30 21:02:59,542 [shard 0:main] rpc - transport.cc:400 - could not parse header from server: {host: docker-rp-3, port: 33145}
DEBUG 2024-05-30 21:02:59,542 [shard 0:main] rpc - transport.cc:400 - could not parse header from server: {host: docker-rp-2, port: 33145}
DEBUG 2024-05-30 21:02:59,542 [shard 1:main] rpc - transport.cc:400 - could not parse header from server: {host: docker-rp-3, port: 33145}
```

With a bit of

```
DEBUG 2024-05-30 21:02:59,542 [shard 0:main] rpc - transport.cc:471 - RPC Client to {host: docker-rp-2, port: 33145} probes: { requests_sent: 36, requests_pending: 0, requests_completed: 36, request_errors: 0, request_timeouts: 0, in_bytes: 10335, out_bytes: 13761, connects: 1, connections: 0, connection_errors: 0, read_dispatch_errors: 0, corrupted_headers: 1, server_correlation_errors: 0, client_correlation_errors: 0, requests_blocked_memory: 0 }
```

^^ note the `corrupted_headers` field. Surprisingly well behaved (too much so probably), but nothing works afaict.

Reply:
I don't think this is testing cogency anyway...definition I'm picking up is more like "are all the certs listed in the CRL within its scope". So e.g. 

- The scope of my CRL is "certs that were revoked because the CA was compromised"
- But there's a cert in the CRL that has "unspecified" reason.

Therefore, my CRL is not "cogent"